src/conf.rs

```diff
@@ -19,6 +23,99 @@ pub enum BoomConfigError {
     ConfigFileNotFound,
     #[error("missing key in config")]
     MissingKeyError,
+    #[error("environment variable expansion error")]
+    EnvExpansionError(#[from] ExpandError),
+}
+
+#[derive(thiserror::Error, Debug)]
+pub enum ExpandError {
+    #[error("Missing environment variable '{var_name}' for placeholder '{placeholder}'")]
+    MissingVariable {
+        var_name: String,
+        placeholder: String,
+    },
+}
+
+/// Loads environment variables from a .env file if it exists.
+/// This function should be called early in the application startup,
+/// typically before any configuration loading.
+///
+/// The function looks for .env files in this order:
+/// 1. .env in the current working directory
+/// 2. .env in the parent directory (useful when running from subdirs)
+/// 3. If none found, continues without error (env vars may be set by system)
+pub fn load_dotenv() {
+    // Try current directory first
+    if std::path::Path::new(".env").exists() {
+        match dotenvy::dotenv() {
+            Ok(_) => info!("Loaded environment variables from .env file"),
+            Err(e) => warn!("Found .env file but failed to load it: {}", e),
+        }
+        return;
+    }
+
+    // Try parent directory (useful when running from subdirectories like api/)
+    if std::path::Path::new("../.env").exists() {
+        match dotenvy::from_path("../.env") {
+            Ok(_) => info!("Loaded environment variables from ../.env file"),
+            Err(e) => warn!("Found ../.env file but failed to load it: {}", e),
+        }
+        return;
+    }
+
+    // No .env file found - this is fine, environment variables may be set by the system
+    debug!("No .env file found, using system environment variables only");
+}
+
+/// Expands environment variable placeholders in a string.
+/// Supports both ${VAR_NAME} and ${VAR_NAME:-default_value} syntax.
+///
+/// Examples:
+/// - "${BOOM_DB_PASSWORD}" -> reads from BOOM_DB_PASSWORD env var
+/// - "${BOOM_DB_PASSWORD:-defaultpass}" -> reads from BOOM_DB_PASSWORD, falls back to "defaultpass"
+fn expand_env_vars(input: &str) -> Result<String, ExpandError> {
+    let re = Regex::new(r"\$\{([^}:]+)(?::-(.*?))?\}").unwrap();
```

Comment:
I wonder if these patterns can't be set at compile time somehow, at the top of the file rather in the functions. With how regex works, in theory stuff can be pre-computed before runtime, but maybe not with this crate? (which is ok, just making sure we don't repeat work we don't need to)

Reply:
We actually shouldn't need this functionality any longer. We'll read from env vars, but won't expand them from their use in the YAML file.